plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java

```diff
@@ -0,0 +1,82 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+
+import com.google.cloud.tools.eclipse.util.io.DeleteAllVisitor;
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+
+public class CleanupOldDeploysJob extends Job {
+
+  private static String NAME = Messages.getString("cleanup.deploy.job.name"); //$NON-NLS-1$
+  private static int OLD_DIRECTORIES_TO_KEEP = 2;
```

Comment:
Yes, that's a better name.

On Tue, Jul 19, 2016, 22:16 Chanseok Oh <notifications@github.com> wrote:

> In
> plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java
> <https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/pull/405#discussion_r71456156>
> :
>
> > +import java.util.SortedSet;
> > +import java.util.TreeSet;
> > +
> > +import org.eclipse.core.runtime.IPath;
> > +import org.eclipse.core.runtime.IProgressMonitor;
> > +import org.eclipse.core.runtime.IStatus;
> > +import org.eclipse.core.runtime.Status;
> > +import org.eclipse.core.runtime.jobs.Job;
> > +
> > +import com.google.cloud.tools.eclipse.util.io.DeleteAllVisitor;
> > +import com.google.cloud.tools.eclipse.util.status.StatusUtil;
> > +
> > +public class CleanupOldDeploysJob extends Job {
> > +
> > +  private static String NAME = Messages.getString("cleanup.deploy.job.name"); //$NON-NLS-1$
> > +  private static int OLD_DIRECTORIES_TO_KEEP = 2;
>
> Ah, my mistake. I see it keeps the two most recent directories. So,
> RECENT_DIRECTORIES_TO_KEEP?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/pull/405/files/df292fb9b34b344ae4ae948b661cf8a4d990da9f#r71456156>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADOO_li98T35kQRgl4x6lBHfmo2MocDPks5qXYUZgaJpZM4JQEAT>
> .
>